common/scala/src/main/scala/whisk/core/entity/Identity.scala

```diff
@@ -74,6 +74,53 @@ object Identity extends MultipleReadersSingleWriterCache[Identity, DocInfo] with
         })
     }
 
+    /**
+      * Many subjects can be belonged to only one namespace
+      * Retrieves a key for namespace + subject.
+      *
+      */
+    def get(datastore: AuthStore, namespace: EntityName, subject: Subject)(
```

Comment:
since you have both the subject and the entity name - does it make more sense to actually look up the doc by subject id (that's a primary key) and then find the namespace in that document?

@markusthoemmes opinion?